src/libraries/Microsoft.Extensions.Configuration.Binder/tests/ConfigurationBinderTests.cs

```diff
@@ -246,6 +246,28 @@ public void CanBindNonInstantiatedIEnumerableWithItems()
             Assert.Equal("Yo2", options.NonInstantiatedIEnumerable.ElementAt(1));
         }
 
+#if NET7_0_OR_GREATER        
+        public class Foo
+        {
+            public void Bar(string s1, string s2)
+            {
+                ArgumentNullException.ThrowIfNull(s1, (object?)s2); // sadly, have to cast to force the correct overload.
```

Comment:
Unfortunately, we have to cast to force the correct overload. This also impacts the parameter name reported in the exception message.